src/Internal/Logger.php

```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * Class Logger
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal;
+
+use WC_Logger;
+use WC_Payments;
+use WCPay\Core\Mode;
+use WCPay\Internal\Proxy\LegacyProxy;
+
+defined( 'ABSPATH' ) || exit; // block direct access.
+
+/**
+ * A wrapper class for interacting with WC_Logger.
+ */
+class Logger {
+
+	const LOG_FILENAME = 'woocommerce-payments';
+
+	/**
+	 * The holding property for our WC_Logger instance.
+	 *
+	 * @var WC_Logger $logger
+	 */
+	private $logger;
+
+	/**
+	 * Legacy proxy.
+	 *
+	 * @var LegacyProxy
+	 */
+	private $legacy_proxy;
+
+	/**
+	 * Mode
+	 *
+	 * @var Mode
+	 */
+	private $mode;
+
+	/**
+	 * Logger constructor.
+	 *
+	 * @param LegacyProxy $legacy_proxy    Legacy proxy.
+	 * @param string      $logger_function Logger function.
```

Comment:
The container will call the function for you, and this class will receive a `WC_Logger` instance, so you can update it. As a matter of fact, here is what happens right now:
<img width="1512" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/5311119/3b9777f4-59f8-4a61-b0e9-208709a3fe6f">